static/js/mouse_util.js

```diff
@@ -0,0 +1,70 @@
+// function to calculate the drag start and end positions and time and
+// determine if the motion was a drag or a normal click.
+function drag() {
+    let start;
+    let time;
+
+    return {
+        start: function (e) {
+            start = { x: e.offsetX, y: e.offsetY };
+            time = new Date().getTime();
+        },
+
+        end: function (e) {
+            const end = { x: e.offsetX, y: e.offsetY };
+
+            let dist;
+            if (start) {
+                // get the linear difference between two coordinates on the screen.
+                dist = Math.sqrt(Math.pow(end.x - start.x, 2) + Math.pow(end.y - start.y, 2));
+            } else {
+                // this usually happens if someone started dragging from outside of
+                // a message and finishes their drag inside the message. The intent
+                // in that case is clearly to select an area, not click a message;
+                // setting dist to Infinity here will ensure that.
+                dist = Infinity;
+            }
+
+            this.val = dist;
+            this.time = new Date().getTime() - time;
+
+            start = undefined;
+
+            return dist;
+        },
+        // function to distinguish between hasty clicks and drag(used for
+        // selecting text). We handle this issue by treating it as a click
+        // if distance is very small (covering the long-click case), or
+        // fairly small and over a short time (covering the hasty click case).
+        // This seems to work nearly perfectly. Once we no longer need to
+        // support older browsers, we may be able to use the window.selection
+        // API instead.
```

Comment:
Will try this. Thanks.

Reply:
The point of this change was so you can now remove all the unnecessary code for tracking distance and time. The unused `start` and `end` can be deleted, the object can just be a function since it no longer has any state, and the ~three lines that will be left are likely not worth putting in their own module.

Reply:
> The point of this change was so you can now remove all the unnecessary code for tracking distance and time. The unused `start` and `end` can be deleted, the object can just be a function since it no longer has any state, and the ~three lines that will be left are likely not worth putting in their own module.

Yeah. I thought about that too, but wasn't sure. Thanks. Will do the changes.